recipe_scrapers/sallysblog.py

```diff
@@ -41,3 +42,18 @@ def ingredients(self):
 
     def author(self):
         return "Sally's Blog"
+
+    def ingredient_groups(self):
+        groups = group_ingredients(
+            self.ingredients(),
+            self.soup,
+            ".shop-studio-recipes-recipe-detail-tabs-description-ingredients__content__ingredient-list__title",
+            ".shop-studio-recipes-recipe-detail-tabs-description-ingredients__content__ingredient-list__ingredient",
+        )
+
+        schema_title = self.schema.title()
+
+        if len(groups) == 1 and schema_title.lower() == groups[0].purpose.lower():
+            return [IngredientGroup(ingredients=self.ingredients(), purpose=None)]
+
+        return groups
```

Comment:
Maybe a small detail, but if there is only a single grouping, I think we could always omit it:

```suggestion
        if len(groups) > 1:
            return groups
        else:
            return [IngredientGroup(ingredients=self.ingredients(), purpose=None)]
```